orange/src/main/java/com/hartwig/hmftools/orange/algo/sigs/SigsInterpreter.java

```diff
@@ -0,0 +1,33 @@
+package com.hartwig.hmftools.orange.algo.sigs;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.hartwig.hmftools.datamodel.sigs.ImmutableSignatureAllocation;
+import com.hartwig.hmftools.datamodel.sigs.SignatureAllocation;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class SigsInterpreter
+{
+    @Nullable
+    public static List<SignatureAllocation> convertAndAnnotateWithEtiology(
+            @Nullable List<com.hartwig.hmftools.common.sigs.SignatureAllocation> signatureAllocation,
+            @NotNull Map<String, String> etiologyPerSignature)
+    {
+        if(signatureAllocation == null)
+        {
+            return null;
+        }
+        return signatureAllocation.stream()
+                .map(signature -> ImmutableSignatureAllocation.builder()
+                        .signature(signature.signature())
+                        .etiology(etiologyPerSignature.getOrDefault(signature.signature(), "-"))
```

Comment:
I would think this is fine, but you may also want to consider adding some extra safeties. Imagine Sigs changes in the meantime and suddenly writes "sig1" rather than "Sig1". Would you want to get "-" etiology without being aware? 

You could also consider mapping MISALLOC to "-", and raise a warning in case any other signature is missing from the map? 

I don't have a super strong opinion though, could go either way